wallet-new/docs/updates.md

```diff
@@ -0,0 +1,563 @@
+clarity):
+
+    data WalletSoftwareUpdate = WalletSoftwareUpdate
+        { wsuSoftwareVersion :: !SoftwareVersion
+        , wsuBlockVesion     :: !BlockVersion
+        , wsuScriptVersion   :: !ScriptVersion
+        , wsuImplicit        :: !Bool
+        , wsuVotesFor        :: !Int
+        , wsuVotesAgainst    :: !Int
+        , wsuPositiveStake   :: !CCoin
+        , wsuNegativeStake   :: !CCoin
+        , wsuSeverity        :: !UpdateSeverity
+        -- ^ The severity of the update (i.e. optional or mandatory).
+        , wsuUpdateState     :: !UpdateState
+        -- ^ The state of this update from users point of view.
+        , wsuSha256Checksum  :: !Sha256Checksum
+        -- ^ The checksum expected for this installer. It will probably
+        -- need to be passed as part of the `UpdateProposal` (maybe in an attribute?),
+        -- or we could reuse the hash which is part of the name, if possible.
+        } deriving (Eq, Show, Generic, Typeable)
+
+where `UpdateSeverity` is defined as an enumeration indicating how compulsory this update is:
+
+    data UpdateSeverity = Mandatory
+                        | Optional
+
+For hard forks updates we would set the severity to `Mandatory`, for soft forks and other non-critical updates
+we would label an update as `Optional`.
+
+-   `Optional` updates can be rejected by the user, as well as applied to will at a later stage;
+-   `Mandatory` updates will need to be applied immediately and won't allow the user to proceed
+    in using Daedalus.
+
+On the other hand, `wsuUpdateState` records the users' decision towards the update:
+
+    data UpdateState = Applied
+                     -- ^ The update has been fully applied.
+                     | Downloaded !Sha256Checksum !FilePath
+                     -- ^ The update has been downloaded but not applied yet. A Checksum
+                     -- can be stored to verify the integrity of the download.
+                     | Skipped
+                     -- ^ The update has been skipped by the user.
+                     | Proposed
+                     -- ^ The update has been just proposed and is waiting for users' feedback.
+
+## Determining the severity<a id="sec-5-2" name="sec-5-2"></a>
+
+For the proposed schema to succeed, we would need a function like the following:
+
+    updateSeverity :: UpdateProposal -> UpdateSeverity
+
+Such function would gather evidence from an `UpdateProposal` that this is a non-negotiable update.
+
+## User stories<a id="sec-5-3" name="sec-5-3"></a>
+
+In order to check the validity of the proposed plan, let's try to run trough some of the common
+user stories for the update system.
+
+-   (1) As a Daedalus user, I want the wallet to prompt me if there is a new update available.
+
+At startup Daedalus would make a call to `GET /api/v1/update` to fetch the last available update. Note
+how this endpoint would **always** return a `WalletSoftwareUpdate`.
```

Comment:
If there were no updates since Daedalus was first installed, then how?
We might say that "well, there already was updates in history of mainnet anyway" and traverse the blockchain in order to find such old updates, but this seems ugly and I believe it's better to just change how API behaves.

I'm not sure why we really need to display here update which was already applied. Why Daedalus would need this information? Probably in order to display information about this last applied update to the user; but it seems to me that it would be more clear to perform this via separate endpoint (like `GET /api/v1/update/history`), which may return full history of already applied updates.